### PR TITLE
[AP-835] Import specific taps

### PR DESCRIPTION
**Pull request by amofakhar**
## Problem

Currently, PipelineWise import command runs on an entire folder and imports all taps.

## Proposed changes

Implementing `taps` option for `import` command to let user to choose specific taps for importing.

## Types of changes

What types of changes does your code introduce to PipelineWise?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] Description above provides context of the change
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Unit tests for changes (not needed for documentation changes)
- [x] CI checks pass with my changes
- [ ] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [x] Commit message/PR title starts with `[AP-NNNN]` (if applicable. AP-NNNN = JIRA ID)
- [x] Branch name starts with `AP-NNN` (if applicable. AP-NNN = JIRA ID)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions

### CODE REVIEW

File: tests/units/cli/test_config.py [hunk 1]
```diff
@@ -359,6 +349,128 @@ def test_save_config(self):
         # Delete the generated JSON config directory
         shutil.rmtree(json_config_dir)
 
+    def test_save_config_selected_tap(self):
+        """Test config target and tap JSON save functionalities if specific taps are selected"""
+        json_config_dir = './pipelinewise-test-config'
+        config = self._get_config(json_config_dir, yaml_path='test_yaml_config_selected_tap')
```
Comment (Samira-El) [inline tests/units/cli/test_config.py:355]: I'm wondering why the need for a new folder with taps `test_yaml_config_selected_tap` and not just use the existing ones under `test_import_command`?

**Review by Samira-El — COMMENTED** [review]
It appears that when `--taps` is used and one of the given taps doesn't exist, nothing happens.
Can we perhaps log a warning in case a given tap id doesn't exist?

File: pipelinewise/cli/__init__.py [hunk 2]
```diff
@@ -208,6 +208,7 @@ def main():
     parser.add_argument('command', type=str, choices=COMMANDS)
     parser.add_argument('--target', type=str, default='*', help='"Name of the target')
     parser.add_argument('--tap', type=str, default='*', help='Name of the tap')
+    parser.add_argument('--taps', type=str, default='*', help='List of taps to import config')
```
[the remark this answers is not in the document]
Comment (amofakhar) [reply pipelinewise/cli/__init__.py]: - [ ] Fix it

File: tests/units/cli/test_config.py (hunk 1 above)
Comment (amofakhar) [reply tests/units/cli/test_config.py]: - [ ] fix it

**Comment by amofakhar** [comment]
> It appears that when `--taps` is used and one of the given taps doesn't exist, nothing happens.
> Can we perhaps log a warning in case a given tap id doesn't exist?

- [ ] Fix it

File: pipelinewise/cli/pipelinewise.py [hunk 3]
```diff
@@ -1613,12 +1614,24 @@ def import_project(self):
...
             total_targets += 1
-            total_taps += len(target.get('taps'))
+            selected_taps = []
+
+            if selected_taps_id == ['*']:
+                total_taps += len(target.get('taps'))
+                selected_taps = target.get('taps')
+
+            else:
+                total_taps = len(selected_taps_id)
```
Comment (Samira-El) [inline pipelinewise/cli/pipelinewise.py:1630]: This variable will be set in every iteration through the targets, I think it's better to move it outside the `for target in config.targets.values():` loop

**Review by Samira-El — COMMENTED** [review]
lgtm, left one last comment

File: pipelinewise/cli/pipelinewise.py (hunk 3 above)
Comment (amofakhar) [reply pipelinewise/cli/pipelinewise.py:1630]: - [ ] Fix it!

**Review by Samira-El — APPROVED** [review]

**Pull request merged**